### PR TITLE
PLFM-5604

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/file/ExternalGoogleCloudUploadDestination.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"description": "The upload destination contains information to start an upload to different types of destinations.",
+	"implements": [
+		{
+			"$ref": "org.sagebionetworks.repo.model.file.UploadDestination"
+		}
+	],
+	"properties": {
+		"endpointUrl": {
+			"type": "string",
+			"description": "the s3 endpoint url (default: https://s3.amazonaws.com)"
```
Comment (nickgros) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/file/ExternalGoogleCloudUploadDestination.json]: fix

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/file/S3FileHandleInterface.json [hunk 2]
```diff
@@ -3,17 +3,7 @@
 	"type": "interface",
 	"implements": [
 		{
-			"$ref": "org.sagebionetworks.repo.model.file.FileHandle"
+			"$ref": "org.sagebionetworks.repo.model.file.CloudProviderFileHandleInterface"
 		}
-	],
-	"properties": {
```
Comment (nickgros) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/file/S3FileHandleInterface.json]: extracted to CloudProviderFileHandleInterface

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOFileHandleDaoImpl.java [hunk 3]
```diff
@@ -287,9 +292,9 @@ public FileHandleResults getAllFileHandles(Iterable<String> ids, boolean include
 	}
 
 	@Override
-	public long getS3objectReferenceCount(String bucketName, String key) {
+	public long getNumberOfReferencesToFile(String metadataType, String bucketName, String key) {
```
Comment (nickgros) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOFileHandleDaoImpl.java:299]: This method counts the number of filehandles that point to the same underlying file. The changes allow the method to work with both S3 and Google Cloud file handles.

File: services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImplTest.java [hunk 4]
```diff
@@ -1,13 +1,15 @@
 package org.sagebionetworks.repo.manager.file;
```
Comment (nickgros) [inline services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImplTest.java:1]: Mostly porting to JUnit5, some added tests

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/file/ExternalGoogleCloudUploadDestination.json (hunk 1 above)
Comment (john-hill) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/file/ExternalGoogleCloudUploadDestination.json]: Upload destination are to be removed.

**Pull request closed without merging**